devel/folly/Portfile

```diff
@@ -49,3 +49,7 @@ variant jemalloc description {Use je_malloc} {
     configure.args-append    -DFOLLY_HAVE_WEAK_SYMBOLS=ON
     configure.ldflags-append -ljemalloc
 }
+
+variant sse42 description {Use SSE4.2 instruction set} {
+    configure.cxxflags-append -msse4.2
+}
```

Comment:
Why is this a variant and not just always done? No dependencies are being added.

Reply:
I made it a variant because it needs hardware support and I was not sure all macs had it. If you tell me they all do, I'll modify the PR.

Reply:
I don't know. You can look at the `atlas` Portfile for guidance, however, because it appears to account it.

Reply:
I made it a default variant when the hardware supports it, as apparently not all mac CPUs do (though all the recent ones do).

Reply:
This is not the right way to handle this sort of situation.

The problem is the build machine (buildbot) might not be the same as the deployment machine, so checking what the build machine supports does not really tell you much.

The way we have handled this sort of situation in other places is to add a non default ‘native’ variant, which passes to the build option -march=native in order to enable the best support for the machine it is being built on. The variant need to be non default as it then only makes sense to force the user to build from source on their local machine when using this variant. Finally, add some port notes telling the user they might want to use this variant, when it is not already enabled.